include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+#ifndef _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+#define _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+
+#include <type_traits>
+#include "nbl/asset/utils/IGLSLCompiler.h" // asset::IGLSLCompiler::E_SPIRV_VERSION
+
+
+namespace nbl::video
+{
+
+struct SPhysicalDeviceLimits
+{
+    /* Vulkan 1.0 Core  */
+    uint32_t maxImageDimension1D;
+    uint32_t maxImageDimension2D;
+    uint32_t maxImageDimension3D;
+    uint32_t maxImageDimensionCube;
+    uint32_t maxImageArrayLayers;
+    uint32_t maxBufferViewSizeTexels;
+    uint32_t maxUBOSize;
+    uint32_t maxSSBOSize;
+    //uint32_t              maxPushConstantsSize;
+    //uint32_t              maxMemoryAllocationCount;
+    //uint32_t              maxSamplerAllocationCount;
+    //VkDeviceSize          bufferImageGranularity;
+    //VkDeviceSize          sparseAddressSpaceSize;
+    //uint32_t              maxBoundDescriptorSets;
+    //uint32_t              maxPerStageDescriptorSamplers;
+    //uint32_t              maxPerStageDescriptorUniformBuffers;
+    uint32_t maxPerStageDescriptorSSBOs;
+    //uint32_t              maxPerStageDescriptorSampledImages;
+    //uint32_t              maxPerStageDescriptorStorageImages;
+    //uint32_t              maxPerStageDescriptorInputAttachments;
+    //uint32_t              maxPerStageResources;
+    //uint32_t              maxDescriptorSetSamplers;
+    uint32_t maxDescriptorSetUBOs;
+    uint32_t maxDescriptorSetDynamicOffsetUBOs;
+    uint32_t maxDescriptorSetSSBOs;
+    uint32_t maxDescriptorSetDynamicOffsetSSBOs;
+    uint32_t maxDescriptorSetImages;
+    uint32_t maxDescriptorSetStorageImages;
+    //uint32_t              maxDescriptorSetInputAttachments;
+    //uint32_t              maxVertexInputAttributes;
+    //uint32_t              maxVertexInputBindings;
+    //uint32_t              maxVertexInputAttributeOffset;
+    //uint32_t              maxVertexInputBindingStride;
+    //uint32_t              maxVertexOutputComponents;
+    //uint32_t              maxTessellationGenerationLevel;
+    //uint32_t              maxTessellationPatchSize;
+    //uint32_t              maxTessellationControlPerVertexInputComponents;
```

Comment:
[GL_MAX_TESS_CONTROL_INPUT_COMPONENTS](https://opengl.gpuinfo.org/displaycapability.php?name=GL_MAX_TESS_CONTROL_INPUT_COMPONENTS)

but its not exactly the same thing, more like an upper bound (since actual vertex count is unknown to do a division)